fetch-in-react-practice/client/src/components/Todos.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import React, { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+const url = (path) => `${process.env.REACT_APP_BASE_URL}${path}`;
+
+export default function Todos() {
+  const [todos, setTodos] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    async function getToDos() {
+      try {
+        const res = await fetch(url('/api/todos'));
+        const jsonData = await res.json();
+        if (!res.ok) {
+          throw new Error(`error ${res.status}`);
+        } else {
+          setTodos(jsonData);
+        }
+      } catch (err) {
+        setError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    getToDos();
+  }, []);
+
+  function addTodo(newTodo) {
+    async function postTodos(newTodo) {
```

Comment:
There's no need to nest this async function in here since event handlers can be async.